requests/models.py

```diff
@@ -820,6 +827,9 @@ def iter_lines(self, chunk_size=ITER_CHUNK_SIZE, decode_unicode=None, delimiter=
             for line in lines:
                 yield line
 
+            # check if the current chunk ends with '\r'
+            last_chunk_ends_with_cr = chunk.endswith('\r' if decode_unicode else b'\r')
```

Comment:
It would also be good to hoist this into the block that holds the `splitlines` call, as we don't need to do this processing in any other situation.